src/http/websocket_http_client.zig

```diff
@@ -1323,22 +1326,33 @@ pub fn NewWebSocketClient(comptime ssl: bool) type {
                             .extended_payload_length_16 => @as(usize, 2),
                             else => unreachable,
                         };
+
                         // we need to wait for more data
-                        if (data.len == 0) return;
+                        if (data.len == 0) {
+                            return;
+                        }
 
-                        if (data.len < byte_size) {
-                            this.terminate(ErrorCode.control_frame_is_fragmented);
-                            terminated = true;
+                        // copy available payload length bytes to a buffer held on this client instance
+                        const total_received = @min(byte_size - this.payload_length_frame_len, data.len);
+                        @memcpy(this.payload_length_frame_bytes[this.payload_length_frame_len..][0..total_received], data[0..total_received]);
+                        this.payload_length_frame_len += @intCast(total_received);
+                        data = data[total_received..];
+
+                        // short read on payload length - we need to wait for more data
+                        // whatever bytes were returned from the short read are kept in `payload_length_frame_bytes`
+                        if (this.payload_length_frame_len < byte_size) {
```

Comment:
should we return in this condition instead of break? like in the `data.len == 0` condition

Reply:
Either a break or return should properly cleanup the `onData` handler. To keep styling consistent I changed it to a break.